### ISSUE TITLE
CTest broken on Windows

### ISSUE THREAD

**Issue by kyrsjo** [SixTrack/SixTrack#191]
It is currently not possible to compile `SixTestWrapper.cpp` on Windows. This makes testing on Windows more difficult, especially automatic testing.

Furthermore, `run_kill` does also not work on this OS.

**Comment by JamesMolson** [comment]
I also need to get the CMakeLists.txt to copy the files, since currently they are symlinked and windows does not support this.

I might make a second windows test program since I'll have to effectively ifdef out everything that is there currently.

Time to learn the win32 api...

**Comment by JamesMolson** [comment]
So far, it works, for normal (non-CR) builds.
Extra checks fail due to the windows \r in new lines.
There may be some horrible things going on with windows path names vs unix.

First to be fixed: CR

**Comment by JamesMolson** [comment]
Path names fixed.
'\r' characters are now removed from files in extra_checks in a pre-pass.
Just CR left to do.

```
# ctest -L fast -j 4
Test project /home/jamolson/build
      Start 13: dipedge
      Start 15: dynk_globalvars
      Start 31: thick4
      Start 30: s316
 1/24 Test #15: dynk_globalvars ..................   Passed   16.24 sec
      Start 25: lostnumxv
 2/24 Test #30: s316 .............................   Passed   21.69 sec
      Start 10: bb
 3/24 Test #31: thick4 ...........................   Passed   26.61 sec
      Start 27: notilt
 4/24 Test #25: lostnumxv ........................   Passed   10.48 sec
      Start 24: lost
 5/24 Test #10: bb ...............................   Passed    9.80 sec
      Start 11: bb_ntwin1
 6/24 Test #24: lost .............................   Passed   11.14 sec
 7/24 Test #27: notilt ...........................   Passed   11.85 sec
      Start 26: lostnotilt
      Start 22: javier
 8/24 Test #11: bb_ntwin1 ........................   Passed   10.32 sec
      Start 19: exact
 9/24 Test #22: javier ...........................   Passed    5.16 sec
      Start 32: thick6dblocks
10/24 Test #19: exact ............................   Passed    2.54 sec
      Start 16: elensidealthin4d
11/24 Test #32: thick6dblocks ....................   Passed    1.93 sec
      Start 34: elensidealthck4d
12/24 Test #26: lostnotilt .......................   Passed    8.37 sec
      Start 35: elensidealthck6d
13/24 Test #16: elensidealthin4d .................   Passed    3.43 sec
      Start  3: SelfTest_compf10
14/24 Test  #3: SelfTest_compf10 .................   Passed    0.24 sec
      Start  5: SelfTest_checkf1014
15/24 Test  #5: SelfTest_checkf1014 ..............   Passed    0.14 sec
      Start  1: SelfTest_verify10
16/24 Test  #1: SelfTest_verify10 ................   Passed    0.29 sec
17/24 Test #34: elensidealthck4d .................   Passed    2.89 sec
      Start  6: SelfTest_read90
      Start  2: SelfTest_checkf10
18/24 Test #35: elensidealthck6d .................   Passed    1.98 sec
19/24 Test  #6: SelfTest_read90 ..................   Passed    0.03 sec
20/24 Test  #2: SelfTest_checkf10 ................   Passed    0.14 sec
      Start  4: SelfTest_checkf110
      Start 12: crabamp
      Start 37: thick6ddynk
21/24 Test  #4: SelfTest_checkf110 ...............   Passed    0.28 sec
22/24 Test #12: crabamp ..........................   Passed    0.50 sec
23/24 Test #37: thick6ddynk ......................   Passed    0.89 sec
24/24 Test #13: dipedge ..........................   Passed   56.94 sec

100% tests passed, 0 tests failed out of 24

Label Time Summary:
fast    = 203.90 sec (24 tests)
tool    =   1.13 sec (6 tests)

Total Test time (real) =  56.97 sec
```

**Comment by JamesMolson** [comment]
CR now seems to work in wine at least as expected. Some weird timing going on...
I'm going to try it on a more real windows machine now.
```
 1/24 Test  #2: SelfTest_checkf10 ................   Passed    0.49 sec
 2/24 Test  #3: SelfTest_compf10 .................   Passed    0.48 sec
 3/24 Test  #1: SelfTest_verify10 ................   Passed    0.49 sec
 4/24 Test  #6: SelfTest_read90 ..................   Passed    1.07 sec
 5/24 Test  #4: SelfTest_checkf110 ...............   Passed    1.17 sec
 6/24 Test #26: lostnotilt .......................   Passed   81.29 sec
 7/24 Test #27: notilt ...........................   Passed  102.81 sec
 8/24 Test #37: thick6ddynk ......................   Passed  126.83 sec
 9/24 Test #25: lostnumxv ........................***Failed  139.07 sec
10/24 Test #32: thick6dblocks ....................   Passed  139.55 sec
11/24 Test #30: s316 .............................   Passed  149.07 sec
12/24 Test #35: elensidealthck6d .................   Passed  154.27 sec
13/24 Test #34: elensidealthck4d .................   Passed  159.18 sec
14/24 Test #16: elensidealthin4d .................   Passed  159.50 sec
15/24 Test #10: bb ...............................   Passed  233.23 sec
16/24 Test #11: bb_ntwin1 ........................   Passed  234.23 sec
17/24 Test #19: exact ............................   Passed  249.83 sec
18/24 Test #22: javier ...........................   Passed  301.72 sec
19/24 Test #31: thick4 ...........................   Passed  361.89 sec
20/24 Test #24: lost .............................   Passed  443.33 sec
21/24 Test #15: dynk_globalvars ..................   Passed  511.71 sec
22/24 Test #13: dipedge ..........................   Passed  676.84 sec
23/24 Test #12: crabamp ..........................   Passed  679.71 sec
24/24 Test  #5: SelfTest_checkf1014 ..............   Passed  679.68 sec

96% tests passed, 1 tests failed out of 24

Label Time Summary:
fast    = 5587.45 sec (24 tests)
tool    = 683.38 sec (6 tests)

Total Test time (real) = 679.72 sec

The following tests FAILED:
	 25 - lostnumxv (Failed)
Errors while running CTest
```

**Comment by JamesMolson** [comment]
Mingw64:
```
      Start  1: SelfTest_verify10
      Start  2: SelfTest_checkf10
      Start  3: SelfTest_compf10
      Start  4: SelfTest_checkf110
 1/37 Test  #4: SelfTest_checkf110 ...............   Passed    0.03 sec
 2/37 Test  #3: SelfTest_compf10 .................   Passed    0.05 sec
 3/37 Test  #2: SelfTest_checkf10 ................   Passed    0.06 sec
 4/37 Test  #1: SelfTest_verify10 ................   Passed    0.08 sec
      Start  5: SelfTest_checkf1014
      Start  6: SelfTest_read90
      Start  7: bbe51
      Start  8: bbe52
 5/37 Test  #6: SelfTest_read90 ..................   Passed    0.05 sec
 6/37 Test  #5: SelfTest_checkf1014 ..............   Passed    0.32 sec
      Start  9: bbe571ib0
      Start 10: bb
 7/37 Test #10: bb ...............................   Passed   28.38 sec
      Start 11: bb_ntwin1
 8/37 Test #11: bb_ntwin1 ........................   Passed   23.15 sec
      Start 12: crabamp
 9/37 Test #12: crabamp ..........................   Passed    2.43 sec
      Start 13: dipedge
10/37 Test #13: dipedge ..........................   Passed   68.06 sec
      Start 14: distance
11/37 Test  #9: bbe571ib0 ........................   Passed  127.37 sec
      Start 15: dynk_globalvars
12/37 Test #15: dynk_globalvars ..................   Passed   21.34 sec
      Start 16: elensidealthin4d
13/37 Test #16: elensidealthin4d .................   Passed    3.52 sec
      Start 17: elensidealthin6d
14/37 Test #17: elensidealthin6d .................   Passed   29.93 sec
      Start 18: eric
15/37 Test #14: distance .........................   Passed  743.43 sec
      Start 19: exact
16/37 Test #19: exact ............................   Passed    2.68 sec
      Start 20: frs60
17/37 Test #18: eric .............................   Passed  752.42 sec
      Start 21: frs
18/37 Test  #8: bbe52 ............................   Passed  1204.98 sec
      Start 22: javier
19/37 Test  #7: bbe51 ............................   Passed  1208.57 sec
      Start 23: lostevery
20/37 Test #22: javier ...........................   Passed    9.24 sec
      Start 24: lost
21/37 Test #24: lost .............................   Passed   18.70 sec
      Start 25: lostnumxv
22/37 Test #25: lostnumxv ........................***Failed    3.62 sec
      Start 26: lostnotilt
23/37 Test #26: lostnotilt .......................   Passed   17.42 sec
      Start 27: notilt
24/37 Test #27: notilt ...........................   Passed   30.80 sec
      Start 28: prob1
25/37 Test #21: frs ..............................   Passed  644.15 sec
      Start 29: prob3
26/37 Test #23: lostevery ........................   Passed  417.61 sec
      Start 30: s316
27/37 Test #30: s316 .............................   Passed   34.07 sec
      Start 31: thick4
28/37 Test #31: thick4 ...........................   Passed   14.81 sec
      Start 32: thick6dblocks
29/37 Test #32: thick6dblocks ....................   Passed    2.13 sec
      Start 33: tilt
30/37 Test #33: tilt .............................   Passed   88.93 sec
      Start 34: elensidealthck4d
31/37 Test #34: elensidealthck4d .................   Passed    3.37 sec
      Start 35: elensidealthck6d
32/37 Test #35: elensidealthck6d .................   Passed    2.75 sec
      Start 36: fma
33/37 Test #20: frs60 ............................   Passed  1211.04 sec
      Start 37: thick6ddynk
34/37 Test #37: thick6ddynk ......................   Passed    2.44 sec
35/37 Test #36: fma ..............................   Passed  465.62 sec
36/37 Test #29: prob3 ............................   Passed  24304.71 sec
37/37 Test #28: prob1 ............................   Passed  25154.56 sec

97% tests passed, 1 tests failed out of 37

Label Time Summary:
fast      = 289.50 sec (24 tests)
medium    = 6894.04 sec (11 tests)
slow      = 49459.27 sec (2 tests)
tool      =   0.58 sec (6 tests)

Total Test time (real) = 26440.53 sec

The following tests FAILED:
         25 - lostnumxv (Failed)
Performing coverage
```

https://abp-cdash.web.cern.ch/abp-cdash/viewTest.php?onlypassed&buildid=32350

**Comment by JamesMolson** [comment]
Fixed in #199
